lib/scholar/decomposition/truncated_svd.ex

```diff
@@ -0,0 +1,225 @@
+defmodule Scholar.Decomposition.TruncatedSVD do
+  @moduledoc """
+  Dimensionality reduction using truncated SVD (aka LSA).
+
+  This transformer performs linear dimensionality reduction by means of
+  truncated singular value decomposition (SVD).
+  """
+  import Nx.Defn
+
+  @derive {Nx.Container,
+           containers: [
+             :components,
+             :explained_variance,
+             :explained_variance_ratio,
+             :singular_values
+           ]}
+  defstruct [
+    :components,
+    :explained_variance,
+    :explained_variance_ratio,
+    :singular_values
+  ]
+
+  tsvd_schema = [
+    num_components: [
+      default: 2,
+      type: :pos_integer,
+      doc: "Desired dimensionality of output data."
+    ],
+    num_iter: [
+      default: 5,
+      type: :pos_integer,
+      doc: "Number of iterations for randomized SVD solver."
+    ],
+    num_oversamples: [
+      default: 10,
+      type: :pos_integer,
+      doc: "Number of oversamples for randomized SVD solver."
+    ],
+    seed: [
```

Comment:
I think you should pass `key` here, which is of type `{:custom, Scholar.Options, :key}`.
See how it's done in e.g. `Scholar.Clustering.KMeans`.